### PR TITLE
Pass matched value to otherwise

### PR DESCRIPTION
**Pull request by oguimbal**
Hi !

Wouldnt it be relevant to pass the matched value to the `.otherwise()` handler ?

before:
```typescript
const value = computeValue();
const result = match(value)
        .when(a, d => d * 2)
        .when(b, d => d / 2)
        .otherwise(() => value + 2);
```

after:
```typescript
const result = match(computeValue())
        .when(a, d => d * 2)
        .when(b, d => d / 2)
        .otherwise(d => d + 2);
```

### CODE REVIEW

**Comment by m-rutter** [comment]
What is the advantage of this Vs just using the value in the outer scope? Isn't it the same value?

**Comment by oguimbal** [comment]
That's just for convenience... it allows to avoid a declaration, making the overall style a bit more functional.

This is especially usefull when using ts-pattern "inline", like I do in react code:

```tsx

const someAtom = atom(...);

const Component = () => <div>match(useRecoilValue(someAtom))
                                                   .when('loading', () => <div>Loading</div>)
                                                   .when(..., () => ...)
                                                   .otherwise(d => <div>Unhandled case: {d}</div>)
                                             </div>
```

File: tests/when.test.ts [hunk 1]
```diff
@@ -197,4 +197,12 @@ describe('when', () => {
...
         .run();
   });
+
+
+  it('should pass matched value to otherwise', () => {
+    const result = match(42)
+      .with(51, d => d)
+      .otherwise(d => d);
+    expect(result).toBe(42);
+  })
```
Comment (gvergnaud) [inline tests/when.test.ts:207]: Since this is unrelated to the `match(...).when` method, I think it would be better to put this test in a separate `otherwise.test.ts` file

**Review by gvergnaud — COMMENTED** [review]
This makes sense and the diff looks ok. Could you please update the documentation in the README in this PR as well before we merge it?

**Comment by oguimbal** [comment]
Hi !

I dont know if that's what you had in mind for the README, but it is done (and I created a brand new `otherwise.test.ts` file)

**Comment by gvergnaud** [comment]
Hey! I meant editing the API reference section in the readme, but don't worry about I'll merge this and take care of it.
